index.js

```diff
@@ -703,135 +703,172 @@
 			});
 		}
 
+		/**
+		 *  Helper function to log amorphic errors.
+		 * @param {*} logger
+		 * @param {*} activity
+		 * @param {*} message
+		 * @param {*} logType
+		 * @param {*} logString
+		 */
+		function postCallErrorLog(logger, activity, message, logType, logString) {
+			let logBody = {
+				component: 'semotus',
+				module: 'processCall.failure',
+				data: {
+					call: remoteCall.name,
+					callTime: logTime(),
+					sequence: remoteCall.sequence
+				}
+			};
+
+			logBody.activity = activity;
+
+			if (logger.data) {
+				logBody.data.message = message;
+			}
+
+			logger[logType](logBody, logString);
+		}
+
+		/**
+		 * Helper function to identify if there's a postServerErrorHandler callback on the base controller
+		 * If there is, we execute the handler, and if we catch an error in the handler, we propogate it up to the logger.
+		 * @param {*} controller
+		 */
+		function resolveErrorHandler(logger, controller, type, remoteCall, remoteCallId, callContext, changeString) {
+			if (controller && controller['postServerErrorHandler']) {
+				let errorType = type;
+				let functionName = remoteCall.name;
+				let obj = undefined;
+				if (session.objects[remoteCall.id]) {
+					obj = session.objects[remoteCall.id];
+				}
+				let logBody = {
+					component: 'semotus',
+					module: 'processCall.failure',
+					activity: 'postCall.resolveErrorHandler',
+					data: {
+						call: remoteCall.name
+					}
+				};
+
+				return Promise.resolve()
+					.then(
+						controller['postServerErrorHandler'].bind(
+							controller,
+							errorType,
+							remoteCallId,
+							obj,
+							functionName,
+							callContext,
+							changeString
+						)
+					)
+					.then(
```

Comment:
if we're not planning on implementing the success case, should this whole then block just be a `.catch`?

Reply:
Can't, we need to eventually remove Q.